OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -214,7 +214,8 @@ public override bool Tick(Actor self)
 
 						if (b != WVec.Zero && c != WVec.Zero)
 						{
-							var cosA = (int)(1024 * (b.LengthSquared + c.LengthSquared - a.LengthSquared) / (2 * b.Length * c.Length));
+							// Use longs in the divisor since the multiplication of both integer lengths could overflow
```

Comment:
I'm probably missing something, but can't we just _not_ multiply the lengths?
Like this:
`var cosA = (int)(1024 * (b.LengthSquared + c.LengthSquared - a.LengthSquared) / 2 / b.Length / c.Length);`

Reply:
Good idea, updated.